### ISSUE TITLE
Safir_control fails silently when started on computer without networking

### ISSUE THREAD

**Issue by DonOregano** [SafirSDK/safir-sdk-core#301]
When safir_control is launched on a Linux computer that lacks network interfaces it will fail silently. It appears to hang indefinitely. An error is reported to syslog:

````
CTRL: Caught 'std::exception' exception from io_service.run():   'set_option: No such device'.
````

It appears that this happens when the node configuration expects to use multicast. The exception is thrown when communication tries to join a multicast group, something which is probably not possible on a loopback interface on Linux.

Everything works fine if the use of multicast is disabled.

We need to fail in a more verbose and explicit way when this happens.

**Comment by DonOregano** [comment]
This does not appear to happen on Windows.

**Comment by DonOregano** [comment]
The reason for the apparent hang was that safir_control did not shut down properly after catching an exception. This was fixed as part of the fix for #306.
